### PR TITLE
Refactor style parsing classes

### PR DESCRIPTION
**Pull request by spassarop**
- Change some constructors regarding CSS scanning
- Improve tests for importing stylesheets

### CODE REVIEW

File: src/main/java/org/owasp/validator/css/CssHandler.java [hunk 1]
```diff
@@ -138,23 +135,22 @@ public CssHandler(Policy policy, LinkedList<URI> embeddedStyleSheets,
 	 * 
 	 * @param policy
 	 *            the policy to use
-	 * @param embeddedStyleSheets
-	 *            the queue of stylesheets imported
 	 * @param errorMessages
```
Comment (davewichers) [inline src/main/java/org/owasp/validator/css/CssHandler.java:138]: Now that I think about it, can you move this new parameter to the end of the parameter list? That seems more typically when there are extra/optional parameters. That way the two APIs are identical, except 1 extra parameter at the end.

File: src/main/java/org/owasp/validator/css/CssHandler.java [hunk 2]
```diff
@@ -120,16 +120,13 @@
 	 * 
 	 * @param policy
 	 *            the policy to use
-	 * @param embeddedStyleSheets
```
Comment (davewichers) [inline src/main/java/org/owasp/validator/css/CssHandler.java:123]: So apparently this parameter isn't needed anymore? Even if that is true, removing it from the API is an API breaking change. Instead, can you create a new API that does NOT take this parameter, and then mark the old API deprecated.  We can't simply add/remove parameters from public methods without going through a deprecation process first, as we don't want to break the API.

File: src/main/java/org/owasp/validator/css/CssScanner.java [hunk 3]
```diff
@@ -96,6 +96,18 @@
 
     private static final Pattern p = Pattern.compile(CDATA, Pattern.DOTALL);
```
Comment (davewichers) [inline src/main/java/org/owasp/validator/css/CssScanner.java:97]: While you are working on this, can you add a comment that describes what this variable is for/what it does? And maybe rename 'p' to something a bit more descriptive?

File: src/main/java/org/owasp/validator/css/CssScanner.java [hunk 4]
```diff
@@ -96,6 +96,18 @@
...
 
+    /**
+     * Constructs a scanner based on the given policy.
```
Comment (davewichers) [inline src/main/java/org/owasp/validator/css/CssScanner.java:100]: Can you add a comment to the constructor description that indicates this version of the constructor defaults shouldParseImportedStyles to false (the default) and they should look at the next constructor for a description of what that feature does.

File: src/main/java/org/owasp/validator/css/CssHandler.java (hunk 2 above)
Comment (spassarop) [reply src/main/java/org/owasp/validator/css/CssHandler.java:123]: On it. Of course you will have the spotbug issue we're trying to remove by leaving the old API.
I'm reverting the changes and adding new constructors with the removed parameter.

File: src/main/java/org/owasp/validator/css/CssScanner.java (hunk 3 above)
Comment (spassarop) [reply src/main/java/org/owasp/validator/css/CssScanner.java]: Done

File: src/main/java/org/owasp/validator/css/CssScanner.java (hunk 4 above)
Comment (spassarop) [reply src/main/java/org/owasp/validator/css/CssScanner.java]: Done

Comment (spassarop) [reply src/main/java/org/owasp/validator/css/CssScanner.java]: Should we mark this constructor as deprecated to delete it in the future?

**Pull request merged**